oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/converter/ObjectToMapStringObjectConverter.java

```diff
@@ -54,7 +54,7 @@ public Object convert(Object source, TypeDescriptor sourceType, TypeDescriptor t
 		}
 		Map<?, ?> sourceMap = (Map<?, ?>) source;
 		if (!sourceMap.isEmpty() && sourceMap.keySet().iterator().next() instanceof String) {
```

Comment:
In my opinion this if block can be removed, as it will be covered by ```Map<String, Object> result = new HashMap<>();
		sourceMap.forEach((k, v) -> result.put(k.toString(), v));``` and a new Map should be created in both cases.